src/parse.js

```diff
@@ -1 +1,96 @@
-export default (...args) => new Date(...args)
+/* eslint no-undefined:0 */
+import curry from 'lodash.curry'
+import {DATE_TOKENS, MONTHS} from './helpers/constants'
+import {fromPairs} from './helpers/util'
+
+const _findMonth = curry((list, key) => {
+  const _key = key.slice(0, 3).toLowerCase()
+
+  return list.findIndex(m => m.toLowerCase().slice(0, 3) === _key) + 1
+})
+
+const findMonth = _findMonth(MONTHS)
+const century = () =>
+  Math.floor(new Date().getFullYear() / 100) * 100
+
+// define a parser for each supported date token
+const TOKEN_PARSERS = {
+  'YYYY': ['year', /\d{4}/, Number],
+  'YY': ['year', /\d{2}/, n => Number(n) + century()],
+  'Y': ['year', /\d{1,2}/, n => Number(n) + century()],
+  'MMMM': ['month', /[a-zA-Z]{3,9}/, findMonth],
+  'MMM': ['month', /[a-zA-Z]{3}/, findMonth],
+  'MM': ['month', /\d{2}/, Number],
+  'M': ['month', /\d{1,2}/, Number],
+  'DD': ['day', /\d{2}/, Number],
+  'D': ['day', /\d{1,2}/, Number],
+  'Do': ['day', /\d{1,2}[a-zA-Z]{2}/, n => Number(n.slice(0, n.length - 2))],
+  'HH': ['hour', /\d{2}/, Number],
+  'H': ['hour', /\d{1,2}/, Number],
+  'mm': ['minute', /\d{2}/, Number],
+  'm': ['minute', /\d{1,2}/, Number],
+  'ss': ['second', /\d{2}/, Number],
+  's': ['second', /\d{1,2}/, Number],
+  'SSS': ['millisecond', /\d{3}/, Number],
+  'SS': ['millisecond', /\d{2,3}/, Number],
+  'S': ['millisecond', /\d{1,3}/, Number],
+}
+const parseFormatString = format => format.match(DATE_TOKENS)
+
+// recursively chop off parts of the datestring and parse them based on the syntax object
+const parseDate = (result, datestring, syntax) => {
+  const touple = syntax[0]
```

Comment:
Im a big fan of nitpicking
Naming was quite difficult here, so if you have any ideas for improvement Id like to hear them.
Does the code make sense ?